config.example.wikimedia.yaml

```diff
@@ -450,6 +443,170 @@ templates:
+                  headers:
+                    content-type: application/json
+                  body: '{$.check_request_storage.body}'
+      /{module:graphoid}/v1/request/{format}/{hash}:
+        get:
+          x-request-handler:
+            - check_storage:
+                request:
+                  method: get
+                  uri: /{domain}/sys/key_value/graphoid.{$.request.params.format}/{hash}
+                  headers:
+                    cache-control: '{cache-control}'
+                return_if:
+                  status: '2xx'
+                return:
+                  status: 200
+                  headers: '{$.check_storage.headers}'
+                  body: '{$.check_storage.body}'
+                catch:
+                  status: 404
+            - load_request:
+                request:
+                  uri: /{domain}/sys/post_data/graphoid.request/{hash}
+            - handle_req:
+                request:
+                  method: post
+                  uri: /{domain}/sys/graphoid/v1/handler/{format}
+                  headers:
+                    cache-control: '{cache-control}'
+                    x-resource-location: '{$.request.params.hash}'
+                    content-type: application/json
+                  body: '{$.load_request.body}'
+      /{module:graphoid}/v1/service_request:
+        post:
+          x-request-handler:
+            - png:
+                request:
+                  method: post
+                  uri: http://graphoid-beta.wmflabs.org/www.mediawiki.org/v2/png
+                  headers:
+                    content-type: 'application/json'
+                  body: '{$.request.body}'
+              svg:
+                request:
+                  method: post
+                  uri: http://graphoid-beta.wmflabs.org/www.mediawiki.org/v2/svg
+                  headers:
+                    content-type: 'application/json'
+                  body: '{$.request.body}'
+            - return_result:
+                return:
+                  status: 200
+                  body:
+                    png:
+                      headers: '{$.png.headers}'
+                      body: '{$.png.body}'
+                    svg:
+                      headers: '{$.svg.headers}'
+                      body: '{$.svg.body}'
+      /{module:graphoid}/v1/handler/{format}:
+        post:
+          x-request-handler:
+            - graphoid:
+                request:
+                  method: post
+                  uri: /{domain}/sys/graphoid/v1/service_request
+                  body: '{$.request.body}'
+            - store_png:
+                request:
+                  method: put
+                  uri: /{domain}/sys/key_value/graphoid.png/{$.request.headers.x-resource-location}
+                  headers: '{$$.merge($$.strip($.graphoid.body.png.headers, "content-encoding"), {"x-resource-location": $.request.headers.x-resource-location, "max-age": $.request.headers.max-age})}'
```

Comment:
This content-encoding stripping looks fairly repetitive. Could we strip those headers further down the stack?

It might also save some complexity to have a function to pick some keys from an object. Something like `$.pick($.request.headers, ["x-resource-location","max-age"])`.